### DISCUSSION TITLE
use of pixiJS in chrome extension blocked by unsafe-eval policy

### DISCUSSION THREAD

**Discussion by eballeste** [pixijs/pixijs#9019] [Q&A] [answered]
A new manifest version 3 is being pushed by browsers (mainly Chrome) which will change the way browser extensions behave. There are 2 new security features in version 3 that make using pixi in my app a royal pain in the ass: 
- browser extension's content/page scripts cannot load remote scripts (not so much of a problem, i solved this by loading a downloaded local version of pixi's prebuilt file)
- CSP headers, cannot use `unsafe-eval` anymore. the extension will throw an error if you try to include this rule either through the extension's manifest.json file or as a meta tag in an extension tabbed-content html page. 

My question is, how can I use the `pixi/unsafe-eval` module when using the prebuilt version of the file? Are there any examples anywhere? The module's readme file makes it seem like this has to go through a bundler in order to work.

**Comment by SuperSodaSea** [comment, marked as answer]
Try this:
```html
<!-- PixiJS v6 -->
<script src="https://unpkg.com/pixi.js@6.5.8/dist/browser/pixi.min.js"></script>
<script src="https://unpkg.com/@pixi/unsafe-eval@6.5.8/dist/browser/unsafe-eval.min.js"></script>

<!-- PixiJS v7 -->
<script src="https://unpkg.com/pixi.js@7.0.5/dist/pixi.min.js"></script>
<script src="https://unpkg.com/@pixi/unsafe-eval@7.0.5/dist/unsafe-eval.min.js"></script>
```

(The browser build for @pixi/unsafe-eval@7.0.0-7.0.4 is missing, if you are using these versions, you can just upgrade to 7.0.5.)

**Reply by eballeste** [reply]
hi! thanks!

something seems to be off with version 7, throws the same error:
`Uncaught (in promise) Error: Current environment does not allow unsafe-eval, please use @pixi/unsafe-eval module to enable support.`

version 6 works fine, shows the loaded pixi banner in the console.

**Reply by eballeste** [reply]
my setup is minimal at the moment:

`index.html`
```html

...
<script src="/scripts/vendor/pixijs-7.0.5.js"></script>
<script src="/scripts/vendor/pixijs-unsafe-eval-7.0.5.js"></script>
<script src="/scripts/pages/main/index.js"></script>
...

```

`index.js`
```javascript

((pixi) => {
...
//do something with pixi
...
})(PIXI)

```

**Reply by SuperSodaSea** [reply]
Oh, you need to manually install it for PixiJS 7.0.x:

```js
PIXI.install(PIXI);
```

In PixiJS 7.1.x (still not released yet) you will not need this any more.
